app/services/control_rate_service.rb

```diff
@@ -97,15 +131,19 @@ def bp_query(time)
 
   private
 
+  def quarterly_report?
+    @quarterly_report
+  end
+
   def cache_key
-    "#{self.class}/#{region.model_name}/#{region.id}/#{date_or_range_cache_key}"
+    "#{self.class}/#{region.model_name}/#{region.id}/#{periods.begin.type}_periods/#{periods_cache_key}"
```

Comment:
I'd move the `type_periods` prefix into the `periods_cache_key` method, keeps it contained